test/unit/test_result_iteration.cxx

```diff
@@ -74,8 +75,59 @@ void test_result_iterator_assignment()
 }
 
 
+void check_employee(std::string name, int salary)
+{
+  PQXX_CHECK(name == "x" or name == "y" or name == "z", "Unknown name.");
+  PQXX_CHECK(salary == 1000 or salary == 1200 or salary == 1500, "Unknown salary.");
+}
+
+
+void test_result_for_each()
+{
+  pqxx::connection conn;
+  pqxx::work tx{conn};
+  tx.exec0("CREATE TEMP TABLE employee(name varchar, salary int)");
+  auto fill{pqxx::stream_to::table(tx, {"employee"}, {"name", "salary"})};
+  fill.write_values("x", 1000);
+  fill.write_values("y", 1200);
+  fill.write_values("z", 1500);
+  fill.complete();
+
+  auto const res{tx.exec("SELECT name, salary FROM employee ORDER BY name")};
+
+  // Use for_each with a function.
+  res.for_each(check_employee);
+
+  // Use for_each with a simple lambda.
+  res.for_each([](std::string name, int salary){ check_employee(name, salary); });
+
+  // Use for_each with a lambda closure.
+  std::string names{};
+  int total{0};
+
+  res.for_each([&names, &total](std::string name, int salary){
+    names.append(name);
+    total += salary;
+  });
+  PQXX_CHECK_EQUAL(names, "xyz", "result::for_each did not accumulate names correctly.");
+  PQXX_CHECK_EQUAL(total, 1000 + 1200 + 1500, "Salaries added up wrong.");
+
+  // In addition to regular conversions, you can receive arguments as
+  // string_view, as well as references.
+  names.clear();
+  total = 0;
+  res.for_each([&names, &total](std::string name, int salary){
```

Comment:
Should the `std::string` here be a `std::string_view`?

Reply:
Hurray!  The callback can now also accept reference parameters.

It was a nice little challenge: how do you deduce a tuple's component types, strip the references, and reassemble it back into a tuple type without ever actually receiving the parameter pack?  One more that I solved by applying `decltype` to invocations of an undefined function template on a `std::declval`.  I'm starting to feel like a one-trick pony.  Is there a better way to do this?  None of the stuff I found on the Internet worked.